Fix memory leaks

```
➜  kafka git:(master) ✗ cppcheck kafka
Checking kafka/callbacks.c ...
kafka/callbacks.c:144:9: error: Memory leak: msg [memleak]
        return NULL;
        ^
kafka/callbacks.c:151:9: error: Memory leak: msg [memleak]
        return NULL;
        ^
kafka/callbacks.c:169:9: error: Memory leak: msg [memleak]
        return NULL;
        ^
kafka/callbacks.c:176:9: error: Memory leak: msg [memleak]
        return NULL;
        ^
kafka/callbacks.c:194:9: error: Memory leak: msg [memleak]
        return NULL;
        ^
kafka/callbacks.c:201:9: error: Memory leak: msg [memleak]
        return NULL;
        ^
Checking kafka/callbacks.c: UNUSED...
Checking kafka/callbacks.c: __GNUC__...
Checking kafka/callbacks.c: __LCLINT__...
1/7 files checked 16% done
Checking kafka/common.c ...
Checking kafka/common.c: UNUSED...
Checking kafka/common.c: __GNUC__...
Checking kafka/common.c: __LCLINT__...
2/7 files checked 17% done
Checking kafka/consumer.c ...
Checking kafka/consumer.c: UNUSED...
Checking kafka/consumer.c: __GNUC__...
Checking kafka/consumer.c: __LCLINT__...
3/7 files checked 56% done
Checking kafka/consumer_msg.c ...
Checking kafka/consumer_msg.c: UNUSED...
Checking kafka/consumer_msg.c: __GNUC__...
Checking kafka/consumer_msg.c: __LCLINT__...
4/7 files checked 62% done
Checking kafka/producer.c ...
Checking kafka/producer.c: UNUSED...
Checking kafka/producer.c: __GNUC__...
Checking kafka/producer.c: __LCLINT__...
5/7 files checked 91% done
Checking kafka/queue.c ...
kafka/queue.c:110:9: error: Memory leak: queue [memleak]
        return NULL;
        ^
6/7 files checked 95% done
Checking kafka/tnt_kafka.c ...
Checking kafka/tnt_kafka.c: UNUSED...
Checking kafka/tnt_kafka.c: __GNUC__...
Checking kafka/tnt_kafka.c: __LCLINT__...
7/7 files checked 100% done
```

--- a/kafka/callbacks.c
+++ b/kafka/callbacks.c
@@ -156,6 +156,7 @@ rebalance_msg_t *new_rebalance_revoke_msg(rd_kafka_topic_partition_list_t *revok
 
     pthread_mutex_t lock;
     if (pthread_mutex_init(&lock, NULL) != 0) {
+        free(msg);
         return NULL;
     }
 
@@ -163,6 +164,7 @@ rebalance_msg_t *new_rebalance_revoke_msg(rd_kafka_topic_partition_list_t *revok
 
     pthread_cond_t sync;
     if (pthread_cond_init(&sync, NULL) != 0) {
+        free(msg);
         return NULL;
     }
 
@@ -181,6 +183,7 @@ rebalance_msg_t *new_rebalance_assign_msg(rd_kafka_topic_partition_list_t *assig
 
     pthread_mutex_t lock;
     if (pthread_mutex_init(&lock, NULL) != 0) {
+        free(msg);
         return NULL;
     }
 
@@ -188,6 +191,7 @@ rebalance_msg_t *new_rebalance_assign_msg(rd_kafka_topic_partition_list_t *assig
 
     pthread_cond_t sync;
     if (pthread_cond_init(&sync, NULL) != 0) {
+        free(msg);
         return NULL;
     }
 
@@ -206,6 +210,7 @@ rebalance_msg_t *new_rebalance_error_msg(rd_kafka_resp_err_t err) {
 
     pthread_mutex_t lock;
     if (pthread_mutex_init(&lock, NULL) != 0) {
+        free(msg);
         return NULL;
     }
 
@@ -213,6 +218,7 @@ rebalance_msg_t *new_rebalance_error_msg(rd_kafka_resp_err_t err) {
 
     pthread_cond_t sync;
     if (pthread_cond_init(&sync, NULL) != 0) {
+        free(msg);
         return NULL;
     }
 

--- a/kafka/queue.c
+++ b/kafka/queue.c
@@ -107,6 +107,7 @@ new_queue() {
 
     pthread_mutex_t lock;
     if (pthread_mutex_init(&lock, NULL) != 0) {
+        free(queue);
         return NULL;
     }
 

